src/Microsoft.DotNet.Cli.Utils/Extensions/LockFileExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.DotNet.Tools.Common;
+using NuGet.Packaging;
+using NuGet.ProjectModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    static class LockFileExtensions
+    {
+        public static string GetPackageDirectory(this LockFile lockFile, LockFileTargetLibrary library)
+        {
+            var packageFolders = lockFile.GetNormalizedPackageFolders();
+
+            var packageFoldersCount = packageFolders.Count();
+            var userPackageFolder = packageFoldersCount == 1 ? string.Empty : packageFolders.First();
```

Comment:
I might be forgetting some context on why it was done this way, but what I would expect here is:

```csharp
FallbackPackagePathResolver resolver = null;

if (packageFoldersCount > 0)
{
  // use packageFolders from the assets file
  var userPackageFolder = packageFolders.FirstOrDefault();
  var fallbackPackageFolders = packageFolders.Skip(1);
  resolver  = new FallbackPackagePathResolver(userPackageFolder, fallbackPackageFolders);
}
else
{
  // assets file does not contain packageFolders, use default settings
  // alternatively this could throw with an error
  var pathContext = new NuGetPathContext(NullSettings.Instance);
  resolver = new FallbackPackagePathResolver(pathContext);
}



```